portal-impl/src/com/liferay/portlet/asset/service/permission/AssetPermission.java

```diff
@@ -30,20 +33,29 @@
 
```

Comment:
This class looks good, but I think it should be left as deprecated and two new classes should be created: AssetCategoriesPermission and AssetTagsPermission (note that categories and tags are plural).

That way we keep the common pattern that each resource has its class. What do you think?